src/HeaderList.php

```diff
@@ -7,15 +7,23 @@ class HeaderList
     /** @var array */
     public $names;
 
+    /** @var bool */
+    public $headersSpecified = false;
+
     /**
      * @param array $names
      */
-    public function __construct(array $names)
+    public function __construct(array $names = null)
     {
-        $this->names = array_map(
-            [$this, 'normalize'],
-            $names
-        );
+        if (is_null($names)) {
+            $this->names = ['date'];
```

Comment:
Is this the right place for this logic? Seems this to me this is a pure data model class, and this logic probably belongs in the signature part of the code?

Reply:
Resolved in https://github.com/99designs/http-signatures-php/pull/42/commits/5a5928c35fde1918a6f3320de3ed98f7845f65c4